demos/_assets/js/jqm-demos.js

```diff
@@ -315,24 +315,40 @@ $( document ).on( "mobileinit", function() {
 				input = this.element.prev("form").find( "input" );
 
 			if ( e.which === $.ui.keyCode.DOWN ) {
-				if ( this.element.find( "li.ui-button-active" ).length === 0 ) {
-					this.element.find( "li:first" ).toggleClass( "ui-button-active" ).find("a").toggleClass( "ui-button-active" );
+				if ( this.element.find( "li.ui-listview-item-active" ).length === 0 ) {
+					this.element.find( "li:not('.ui-screen-hidden'):first" )
+					.toggleClass( "ui-listview-item-active" )
+					.find("a").toggleClass( "ui-button-active" );
 				} else {
-					this.element.find( "li.ui-button-active a" ).toggleClass( "ui-button-active");
-					this.element.find( "li.ui-button-active" ).toggleClass( "ui-button-active" ).next().toggleClass( "ui-button-active" ).find("a").toggleClass( "ui-button-active" );
+					this.element.find( "li.ui-listview-item-active a" )
+					.toggleClass( "ui-button-active");
+
+					this.element.find( "li.ui-listview-item-active" )
+					.toggleClass( "ui-listview-item-active" )
+					.nextAll("li:not('.ui-screen-hidden'):first")
+					.toggleClass( "ui-listview-item-active" )
+					.find("a").toggleClass( "ui-button-active" );
 				}
-
-				this.highlightDown();
 			} else if ( e.which === $.ui.keyCode.UP ) {
-				if ( this.element.find( "li.ui-button-active" ).length !== 0 ) {
-					this.element.find( "li.ui-button-active a" ).toggleClass( "ui-button-active");
-					this.element.find( "li.ui-button-active" ).toggleClass( "ui-button-active" ).prev().toggleClass( "ui-button-active" ).find("a").toggleClass( "ui-button-active" );
+				if ( this.element.find( "li.ui-listview-item-active" ).length !== 0 ) {
+					this.element.find( "li.ui-listview-item-active a" )
+					.toggleClass( "ui-button-active");
+
+					this.element.find( "li.ui-listview-item-active" )
+					.toggleClass( "ui-listview-item-active" )
+					.prevAll("li:not('.ui-screen-hidden'):first")
+					.toggleClass( "ui-listview-item-active" )
+					.find("a").toggleClass( "ui-button-active" );
 				} else {
-					this.element.find( "li:last" ).toggleClass( "ui-button-active" ).find("a").toggleClass( "ui-button-active" );
+					this.element.find( "li:not('.ui-screen-hidden'):last" )
+					.toggleClass( "ui-listview-item-active" )
+					.find("a").toggleClass( "ui-button-active" );
 				}
-				this.highlightUp();
-			} else if ( typeof e.which !== "undefined" ) {
-				this.element.find( "li.ui-button-active" ).removeClass( "ui-button-active" );
+			} else if ( e.which === $.ui.keyCode.ENTER ) {
+				this.submitHandler();
+			}else if ( typeof e.which !== "undefined" ) {
```

Comment:
is there a reason we need to do a typeof check here an not just `if ( !e.which )`